### PR TITLE
DOT-2845: Document the list of possible errors for API response

### PR DESCRIPTION
**Pull request by kitro**
https://nextmediama.atlassian.net/browse/DOT-2845

### CODE REVIEW

File: docs/errors/index.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    "code": 0,
+    "detail": "The code field is required.",
+    "meta": []
+}
+```
+
+### Unsupported request type error
+```json
+{
+    "status": 404,
```
Comment (Whyounes) [inline docs/errors/index.md:40]: i don't think this is 404 !

File: docs/errors/index.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+```
+
+### Validation error 
+```json
+{
+    "status": 422,
+    "code": 0,
+    "detail": "The code field is required.",
+    "meta": []
```
Comment (Whyounes) [inline docs/errors/index.md:33]: example of meta

File: docs/errors/index.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+| ----------- | ---------------------------------------------- |
+| 404         | The resource requested not found or deleted |
+| 405         | _Method not allowed_ this error happened when the endpoint requested doesn't support the current request types |
+| 422         | _Unprocessable entity_ this error happened when the request doesn't respect the validation ex: missing required field, invalid fields type|
+| 500         | _500 Internal Server Error_ this error is generic happened when something got wrong in the service side while processing the request  |
+
+
+### Examples
+
+#### Customer not found error
```
Comment (Whyounes) [inline docs/errors/index.md:17]: Just `Not Found`

File: docs/errors/index.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    "detail": "The POST method is not supported for this route. Supported methods: GET, HEAD, PUT, DELETE.",
+    "meta": []
+}
+```
+
+### Internal error
+```json
+{
+    "status": 500,
+    "code": "",
```
Comment (Whyounes) [inline docs/errors/index.md:51]: we should avoid returning the code in all errors, since we don't support this now

File: docs/errors/index.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    "detail": "internal error",
+    "meta": []
```
Comment (Whyounes) [inline docs/errors/index.md:53]: no meta in 500

File: docs/errors/index.md (hunk 1 above)
Comment (kitro) [reply docs/errors/index.md:38]: Yes, when we catch `NotFoundHttpException` this the return type.

File: docs/errors/index.md (hunk 2 above)
Comment (kitro) [reply docs/errors/index.md:31]: The are no example of meta yet

File: docs/errors/index.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+
+### Examples
+
+#### Not found
+```json
+{
+    "status": 404,
+    "detail": "Customer not found",
+    "meta": []
```
Comment (Whyounes) [inline docs/errors/index.md:22]: meta is not needed here as we talked about before

File: docs/errors/index.md (hunk 1 above)
Comment (Whyounes) [reply docs/errors/index.md:50]: the status field won't match the actual HTTP status coed returned, and thus create a confusion for the user

**Pull request merged**